src/Diagnostics.DataProviders/DataProviders/SupportObserverDataProvider.cs

```diff
@@ -72,14 +72,30 @@ public override async Task<dynamic> GetCertificatesInResourceGroupAsync(string s
             return await GetRuntimeSiteSlotMapInternal(stampName, siteName, slotName);
         }
 
-        private async Task<Dictionary<string, List<RuntimeSitenameTimeRange>>> GetRuntimeSiteSlotMapInternal(string stampName, string siteName, string slotName)
+        public override async Task<Dictionary<string, List<RuntimeSitenameTimeRange>>> GetRuntimeSiteSlotMap(OperationContext<App> cxt, string stampName = null, string siteName = null, string slotName = null, DateTime? endTime = null)
+        {
+            if (cxt?.Resource?.Stamp == null)
+                throw new ArgumentNullException(nameof(cxt));
+
+            stampName = string.IsNullOrWhiteSpace(stampName) ? cxt.Resource.Stamp.InternalName : stampName;
+            siteName = string.IsNullOrWhiteSpace(siteName) ? cxt.Resource.Name : siteName;
+            endTime = endTime == null ? DateTime.SpecifyKind(DateTime.Parse(cxt.EndTime), DateTimeKind.Utc) : endTime;
```

Comment:
The OperationContext already has EndTime in UTC format

Reply:
The `endTime` parameter probably should probably be checked. Since you are accepting the OperationContext, you probably don't need these optional parameters right?

Reply:
cxt.EndTime was copypasta, but the optional parameters are there as overrides so you aren't thrown completely out of the `cxt` call if you want a different endTime, or the same siteName on a different stamp